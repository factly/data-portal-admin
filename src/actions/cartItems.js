import axios from '../utils/axios';
import {
  ADD_CARTITEMS,
  SET_CARTITEM_LOADING,
  SET_CARTITEM_REQUEST,
  SET_CARTITEM_IDS,
  CARTITEM_API,
} from '../constants/cartItems';
import { addProducts } from './products';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadCartItems = (id, page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      cartItems: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setCartItemIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${CARTITEM_API(id)}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { id, page: page, limit: limit, ids: currentPageIds };
    dispatch(setCartItemRequest(currentReq, total));
    dispatch(addCartItems(nodes));
    dispatch(setCartItemIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

const setLoading = (loading) => {
  return {
    type: SET_CARTITEM_LOADING,
    payload: { loading },
  };
};

export const addCartItems = (cartItems) => (dispatch) => {
  const products = getValues(cartItems, 'product');
  dispatch(addProducts(products));

  dispatch({
    type: ADD_CARTITEMS,
    payload: {
      cartItems: buildObjectOfItems(deleteKeys(cartItems, ['product'])),
    },
  });
};

const setCartItemRequest = (req, total) => {
  return {
    type: SET_CARTITEM_REQUEST,
    payload: { req, total },
  };
};

const setCartItemIds = (ids) => {
  return {
    type: SET_CARTITEM_IDS,
    payload: { ids },
  };
};
