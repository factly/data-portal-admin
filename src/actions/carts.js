import axios from '../utils/axios';
import {
  baseUrl,
  LOADING_CARTS,
  LOADING_CART_DETAILS,
  LOAD_CARTS_SUCCESS,
  LOAD_CARTS_FAILURE,
  GET_CART_ITEMS_SUCCESS,
  GET_CART_ITEMS_FAILURE,
} from '../constants/carts';

export const loadCarts = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCarts());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadCartsFailure(error.message));
    });

    if (response) {
      dispatch(loadCartsSuccess(response.data));
    }
  };
};

export const getCartItems = (id, page, limit) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}/items`;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    dispatch(loadingCartDetails());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(getCartItemsFailure(error.message));
    });

    if (response) {
      dispatch(getCartItemsSuccess(response.data));
    }
  };
};

const loadingCarts = () => {
  return {
    type: LOADING_CARTS,
  };
};

const loadingCartDetails = () => {
  return {
    type: LOADING_CART_DETAILS,
  };
};

const loadCartsSuccess = (data) => {
  return {
    type: LOAD_CARTS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const loadCartsFailure = (message) => {
  return {
    type: LOAD_CARTS_FAILURE,
    payload: message,
  };
};

const getCartItemsSuccess = (data) => {
  return {
    type: GET_CART_ITEMS_SUCCESS,
    payload: {
      items: data.nodes,
      total: data.total,
    },
  };
};

const getCartItemsFailure = (message) => {
  return {
    type: GET_CART_ITEMS_FAILURE,
    payload: message,
  };
};
