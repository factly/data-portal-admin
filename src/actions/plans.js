import axios from '../utils/axios';
import {
  baseUrl,
  ADD_PLANS_LIST_REQUEST,
  SET_PLANS_LIST_CURRENT_PAGE,
  LOADING_PLANS,
  LOAD_PLANS_SUCCESS,
  SET_PLANS_LIST_TOTAL,
  LOAD_PLANS_FAILURE,
  GET_PLAN_SUCCESS,
  GET_PLAN_FAILURE,
  CREATING_PLAN,
  CREATE_PLAN_SUCCESS,
  CREATE_PLAN_FAILURE,
  UPDATING_PLAN,
  UPDATE_PLAN_SUCCESS,
  UPDATE_PLAN_FAILURE,
  DELETING_PLAN,
  DELETE_PLAN_SUCCESS,
  DELETE_PLAN_FAILURE,
} from '../constants/plans';
import { getIds, buildObjectOfItems } from '../utils/objects';

export const loadPlans = (page, limit) => {
  return async (dispatch, getState) => {
    let url = baseUrl;
    if (page && limit) {
      url = `${url}?page=${page}&limit=${limit}`;
    }

    const {
      plans: { req },
    } = getState();

    let found = false;
    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
        found = true;
      }
    }

    if (found) {
      dispatch(setListCurrentPage(ids));
      return;
    }

    dispatch(loadingPlans());

    const response = await axios({
      url: url,
      method: 'get',
    }).catch((error) => {
      dispatch(loadPlansFailure(error.message));
    });

    if (response) {
      const { nodes, total } = response.data;
      const currentPageIds = getIds(nodes);
      const req = { page: page, limit: limit, ids: currentPageIds };
      dispatch(addListRequest(req));
      dispatch(loadPlansSuccess(nodes));
      dispatch(setListCurrentPage(currentPageIds));
      dispatch(setPlansListTotal(total));
    }
  };
};

export const createPlan = (data) => {
  return async (dispatch, getState) => {
    dispatch(creatingPlan());

    const response = await axios({
      url: baseUrl,
      method: 'post',
      data: data,
    }).catch((error) => {
      dispatch(createPlanFailure(error.message));
    });

    if (response) {
      dispatch(createPlanSuccess(response.data));
    }
  };
};

export const updatePlan = (id, data) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(updatingPlan());

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    }).catch((error) => {
      dispatch(updatePlanFailure(error.message));
    });

    if (response) {
      dispatch(updatePlanSuccess(response.data));
    }
  };
};

export const deletePlan = (id) => {
  return async (dispatch, getState) => {
    let url = `${baseUrl}/${id}`;

    dispatch(deletingPlan());

    const response = await axios({
      url: url,
      method: 'delete',
    }).catch((error) => {
      dispatch(deletePlanFailure(error.message));
    });

    if (response) {
      dispatch(deletePlanSuccess(id));
    }
  };
};

export const getPlan = (id) => {
  return async (dispatch, getState) => {
    const {
      plans: { items },
    } = getState();

    if (items[id]) {
      dispatch(getPlanSuccess({ ...items[id] }));
      return;
    }

    dispatch(loadingPlans());

    const response = await axios({
      url: `${baseUrl}/${id}`,
      method: 'get',
    }).catch((error) => {
      dispatch(getPlanFailure(error.message));
    });

    if (response) {
      dispatch(getPlanSuccess(response.data));
    }
  };
};

const getPlanSuccess = (plan) => {
  return {
    type: GET_PLAN_SUCCESS,
    payload: plan,
  };
};

const getPlanFailure = (message) => {
  return {
    type: GET_PLAN_FAILURE,
    payload: message,
  };
};

const loadingPlans = () => {
  return {
    type: LOADING_PLANS,
  };
};

const setPlansListTotal = (total) => {
  return {
    type: SET_PLANS_LIST_TOTAL,
    payload: total,
  };
};

const addListRequest = (req) => {
  return {
    type: ADD_PLANS_LIST_REQUEST,
    payload: req,
  };
};

const setListCurrentPage = (ids) => {
  return {
    type: SET_PLANS_LIST_CURRENT_PAGE,
    payload: ids,
  };
};

export const loadPlansSuccess = (plans) => {
  return {
    type: LOAD_PLANS_SUCCESS,
    payload: {
      items: buildObjectOfItems(plans),
    },
  };
};

const loadPlansFailure = (message) => {
  return {
    type: LOAD_PLANS_FAILURE,
    payload: message,
  };
};

const creatingPlan = () => {
  return {
    type: CREATING_PLAN,
  };
};

const createPlanSuccess = (plan) => {
  return {
    type: CREATE_PLAN_SUCCESS,
    payload: plan,
  };
};

const createPlanFailure = (message) => {
  return {
    type: CREATE_PLAN_FAILURE,
    payload: message,
  };
};

const updatingPlan = () => {
  return {
    type: UPDATING_PLAN,
  };
};

const updatePlanSuccess = (plan) => {
  return {
    type: UPDATE_PLAN_SUCCESS,
    payload: plan,
  };
};

const updatePlanFailure = (message) => {
  return {
    type: UPDATE_PLAN_FAILURE,
    payload: message,
  };
};

const deletingPlan = () => {
  return {
    type: DELETING_PLAN,
  };
};

const deletePlanSuccess = (id) => {
  return {
    type: DELETE_PLAN_SUCCESS,
    payload: id,
  };
};

const deletePlanFailure = (message) => {
  return {
    type: DELETE_PLAN_FAILURE,
    payload: message,
  };
};
