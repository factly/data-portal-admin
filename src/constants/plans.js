export const LOADING_PLANS = 'LOADING_PLANS';
export const LOAD_PLANS_SUCCESS = 'LOAD_PLANS_SUCCESS';
export const SET_PLANS_LIST_TOTAL = 'SET_PLANS_LIST_TOTAL';
export const LOAD_PLANS_FAILURE = 'LOAD_PLANS_FAILURE';
export const CREATING_PLAN = 'CREATING_PLAN';
export const CREATE_PLAN_SUCCESS = 'CREATE_PLAN_SUCCESS';
export const CREATE_PLAN_FAILURE = 'CREATE_PLAN_FAILURE';
export const UPDATING_PLAN = 'UPDATING_PLAN';
export const UPDATE_PLAN_SUCCESS = 'UPDATE_PLAN_SUCCESS';
export const UPDATE_PLAN_FAILURE = 'UPDATE_PLAN_FAILURE';
export const DELETING_PLAN = 'DELETING_PLAN';
export const DELETE_PLAN_SUCCESS = 'DELETE_PLAN_SUCCESS';
export const DELETE_PLAN_FAILURE = 'DELETE_PLAN_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/plans`;
