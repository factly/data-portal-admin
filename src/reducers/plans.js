import {
  LOADING_PLANS,
  LOAD_PLANS_SUCCESS,
  SET_PLANS_LIST_TOTAL,
  LOAD_PLANS_FAILURE,
  CREATING_PLAN,
  CREATE_PLAN_SUCCESS,
  CREATE_PLAN_FAILURE,
  UPDATE_PLAN_SUCCESS,
  UPDATE_PLAN_FAILURE,
  DELETE_PLAN_SUCCESS,
  DELETE_PLAN_FAILURE,
} from '../constants/plans';
import { unique } from '../utils/objects';

const initialState = {
  list: { loading: false, ids: [], items: {}, total: 0 },
};

export default function plansReducer(state = initialState, action = {}) {
  switch (action.type) {
    case LOADING_PLANS:
      return {
        ...state,
        list: {
          ...state.list,
          loading: true,
        },
      };
    case LOAD_PLANS_SUCCESS: {
      const { ids, items } = action.payload;
      const { list } = state;
      return {
        ...state,
        list: {
          ...list,
          loading: false,
          ids: unique([...list.ids, ...ids]),
          items: { ...list.items, ...items },
        },
      };
    }
    case SET_PLANS_LIST_TOTAL:
      return {
        ...state,
        list: {
          ...state.list,
          total: action.payload,
        },
      };
    case LOAD_PLANS_FAILURE:
      return {
        ...state,
        list: {
          ...state.list,
          loading: false,
        },
      };
    case CREATING_PLAN:
      return {
        ...state,
        list: {
          ...state.list,
          loading: true,
        },
      };
    case CREATE_PLAN_SUCCESS: {
      const { list } = state;
      const plan = action.payload;
      return {
        ...state,
        list: {
          ...list,
          loading: false,
          ids: [...list.ids, plan.id],
          items: { ...list.items, [plan.id]: plan },
          total: list.total + 1,
        },
      };
    }
    case CREATE_PLAN_FAILURE:
      return {
        ...state,
        list: {
          ...state.list,
          loading: false,
        },
      };
    case UPDATE_PLAN_SUCCESS: {
      const { list } = state;
      const plan = action.payload;
      return {
        ...state,
        list: {
          ...list,
          loading: false,
          items: { ...list.items, [plan.id]: plan },
        },
      };
    }
    case UPDATE_PLAN_FAILURE:
      return {
        ...state,
        list: {
          ...state.list,
          loading: false,
        },
      };
    case DELETE_PLAN_SUCCESS: {
      const { list } = state;
      const id = action.payload;
      const index = list.ids.indexOf(id);
      const newIds = [...list.ids];
      newIds.splice(index, 1);
      const newItems = { ...list.items };
      delete newItems[id];
      return {
        ...state,
        list: {
          ...list,
          loading: false,
          ids: newIds,
          items: newItems,
          total: list.total - 1,
        },
      };
    }
    case DELETE_PLAN_FAILURE:
      return {
        ...state,
        list: {
          ...state.list,
          loading: false,
        },
      };
    default:
      return state;
  }
}
