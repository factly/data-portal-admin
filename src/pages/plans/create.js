import React from 'react';
import { useDispatch } from 'react-redux';
import { useHistory } from 'react-router-dom';

import PlanCreateForm from './components/PlanCreateForm';
import { createPlan } from '../../actions/plans';

const PlanCreate = () => {
  const history = useHistory();
  const dispatch = useDispatch();

  const onCreate = async (values) => {
    await dispatch(createPlan(values));
    history.push('/plans');
  };

  return <PlanCreateForm onSubmit={onCreate} />;
};

export default PlanCreate;
