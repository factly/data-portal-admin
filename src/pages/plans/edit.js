import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import PlanCreateForm from './components/PlanCreateForm';
import { getPlan, updatePlan } from '../../actions/plans';

const EditPlan = () => {
  const { id } = useParams();
  const history = useHistory();
  const dispatch = useDispatch();
  const { loading, plan } = useSelector(({ plans }) => ({
    loading: plans.loading,
    plan: plans.items[id],
  }));

  React.useEffect(() => {
    dispatch(getPlan(id));
  }, []);

  if (loading) return <Skeleton />;

  const onUpdate = async (values) => {
    await dispatch(updatePlan(id, values));
    history.push('/plans');
  };

  return <PlanCreateForm data={plan} onSubmit={onUpdate} />;
};

export default EditPlan;
