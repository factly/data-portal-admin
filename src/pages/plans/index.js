import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Table, Input, Button, Popconfirm, Form, notification } from 'antd';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { EditOutlined, DeleteOutlined, SaveOutlined, CloseOutlined } from '@ant-design/icons';

import { loadPlans, updatePlan, deletePlan } from '../../actions/plans';

const EditableCell = ({ editing, dataIndex, title, record, index, children, ...restProps }) => {
  return (
    <td {...restProps}>
      {editing ? (
        <Form.Item
          name={dataIndex}
          style={{
            margin: 0,
          }}
          rules={[
            {
              required: true,
              message: `Please enter ${title}!`,
            },
          ]}
        >
          <Input />
        </Form.Item>
      ) : (
        children
      )}
    </td>
  );
};

const Plans = (props) => {
  const [form] = Form.useForm();
  const [editingKey, setEditingKey] = useState('');
  const { data, total, load, update, remove } = props;

  React.useEffect(() => {
    load();
  }, [load]);

  const get = (page, limit) => {
    cancel();
    load(page, limit);
  };

  const isEditing = (record) => record.id === editingKey;

  const edit = (record) => {
    form.setFieldsValue(record);
    setEditingKey(record.id);
  };

  const cancel = () => {
    setEditingKey('');
  };

  const save = async (key) => {
    try {
      const row = await form.validateFields();
      const index = data.findIndex((item) => item.id === key);

      if (index > -1) {
        update(key, row, index)
          .then(() => {
            setEditingKey('');
            notification.success({
              message: 'Success',
              description: 'Plan succesfully updated',
            });
          })
          .catch(() => {
            notification.error({
              message: 'Error',
              description: 'Something went wrong',
            });
          });
      }
    } catch (err) {
      notification.warning({
        message: 'Warning',
        description: 'Validation failed',
      });
    }
  };

  const deletePlan = (key) => {
    const index = data.findIndex((item) => item.id === key);
    if (index > -1) {
      remove(key, index)
        .then(() => {
          notification.success({
            message: 'Success',
            description: 'Membership succesfully deleted',
          });
        })
        .catch(() => {
          notification.error({
            message: 'Error',
            description: 'Something went wrong',
          });
        });
    }
  };

  const columns = [
    {
      title: 'Name',
      dataIndex: 'plan_name',
      width: '20%',
      editable: true,
    },
    {
      title: 'Info',
      dataIndex: 'plan_info',
      width: '20%',
      editable: true,
    },
    {
      title: 'Status',
      dataIndex: 'status',
      width: '20%',
      editable: true,
    },
    {
      title: 'Created At',
      dataIndex: 'created_at',
      width: '20%',
      render: (_, record) => {
        return <span title={record.created_at}>{moment(record.created_at).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        const editable = isEditing(record);
        return editable ? (
          <span>
            <Button
              type="primary"
              icon={<SaveOutlined />}
              onClick={() => save(record.id)}
              style={{
                marginRight: 8,
              }}
            >
              Save
            </Button>
            <Popconfirm title="Sure to cancel?" onConfirm={cancel}>
              <Button icon={<CloseOutlined />}>Cancel</Button>
            </Popconfirm>
          </span>
        ) : (
          <span>
            <Button
              type="primary"
              icon={<EditOutlined />}
              disabled={editingKey !== ''}
              onClick={() => edit(record)}
              style={{
                marginRight: 8,
              }}
            >
              Edit
            </Button>
            <Popconfirm
              disabled={editingKey !== ''}
              title="Sure to delete?"
              onConfirm={() => deletePlan(record.id)}
            >
              <Button disabled={editingKey !== ''} icon={<DeleteOutlined />}>
                Delete
              </Button>
            </Popconfirm>
          </span>
        );
      },
    },
  ];

  const mergedColumns = columns.map((col) => {
    if (!col.editable) {
      return col;
    }

    return {
      ...col,
      onCell: (record) => ({
        record,
        dataIndex: col.dataIndex,
        title: col.title,
        editing: isEditing(record),
      }),
    };
  });

  return (
    <div>
      <Link to={'/plans/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Plan
        </Button>
      </Link>
      <Form form={form} component={false}>
        <Table
          components={{
            body: {
              cell: EditableCell,
            },
          }}
          bordered
          rowKey="id"
          dataSource={data}
          columns={mergedColumns}
          rowClassName="editable-row"
          pagination={{
            defaultPageSize: 5,
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

Plans.propTypes = {
  data: PropTypes.array.isRequired,
  total: PropTypes.number.isRequired,
  load: PropTypes.func.isRequired,
  update: PropTypes.func.isRequired,
  remove: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => {
  const { list } = state.plans;
  return {
    data: list.items,
    total: list.total,
  };
};

const mapDispatchToProps = (dispatch) => ({
  load: (page, limit) => dispatch(loadPlans(page, limit)),
  update: (id, data, index) => dispatch(updatePlan(id, data, index)),
  remove: (id, index) => dispatch(deletePlan(id, index)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Plans);
