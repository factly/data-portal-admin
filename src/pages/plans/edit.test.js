import React from 'react';
import { useHistory } from 'react-router-dom';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { useDispatch, Provider, useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

import '../../matchMedia.mock';
import EditPlan from './edit';
import * as actions from '../../actions/plans';
import PlanEditForm from './components/PlanCreateForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
  useSelector: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn().mockReturnValue({ id: '1' }),
}));

jest.mock('../../actions/plans', () => ({
  getPlan: jest.fn(),
  addPlan: jest.fn(),
  updatePlan: jest.fn(),
}));

describe('Plans List component', () => {
  const store = mockStore({
    plans: {
      req: [],
      details: {
        '1': {
          id: 1,
          name: 'Plan-1',
          slug: 'plan-1',
          description: 'description',
        },
        '2': {
          id: 2,
          name: 'Plan-2',
          slug: 'plan-2',
          description: 'description',
        },
      },
      loading: true,
    },
  });
  store.dispatch = jest.fn(() => ({}));

  const mockedDispatch = jest.fn();
  useDispatch.mockReturnValue(mockedDispatch);

  describe('snapshot testing', () => {
    it('should render the component', () => {
      useSelector.mockReturnValueOnce({
        plan: {
          id: 1,
          name: 'plan',
          slug: 'slug',
          description: 'description',
        },
        loading: false,
      });
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditPlan />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with empty data', () => {
      useSelector.mockReturnValueOnce({
        plan: {},
        loading: false,
      });
      let component;
      store.details = {};
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditPlan />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match skeleton while loading', () => {
      useSelector.mockReturnValueOnce({
        plan: {},
        loading: true,
      });
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditPlan />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    afterEach(() => {
      wrapper.unmount();
    });
    it('should call get action', () => {
      useSelector.mockReset();
      useSelector.mockReturnValueOnce({ plan: null, loading: true });
      actions.getPlan.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditPlan />
          </Provider>,
        );
      });
      expect(actions.getPlan).toHaveBeenCalledWith('1');
    });
    it('should call updatePlan', (done) => {
      useSelector.mockReset();
      useSelector.mockReturnValueOnce({ plan: {}, loading: false });
      actions.updatePlan.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditPlan />
          </Provider>,
        );
      });
      wrapper.find(PlanEditForm).props().onSubmit({ test: 'test' });
      setTimeout(() => {
        expect(actions.updatePlan).toHaveBeenCalledWith('1', { test: 'test' });
        expect(push).toHaveBeenCalledWith('/plans');
        done();
      }, 0);
    });
  });
});
