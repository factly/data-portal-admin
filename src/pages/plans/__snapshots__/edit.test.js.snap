// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plans List component snapshot testing should match component with empty data 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="plans_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_plan_name"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_plan_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Premium"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_plan_info"
        title="Info"
      >
        Info
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_plan_info"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Allows users to access premium content"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_status"
        title="Status"
      >
        Status
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_status"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Pending"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Plans List component snapshot testing should match skeleton while loading 1`] = `
<div
  className="ant-skeleton"
>
  <div
    className="ant-skeleton-content"
  >
    <h3
      className="ant-skeleton-title"
      style={
        Object {
          "width": "38%",
        }
      }
    />
    <ul
      className="ant-skeleton-paragraph"
    >
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": "61%",
          }
        }
      />
    </ul>
  </div>
</div>
`;

exports[`Plans List component snapshot testing should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="plans_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_plan_name"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_plan_name"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Premium"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_plan_info"
        title="Info"
      >
        Info
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_plan_info"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Allows users to access premium content"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="plans_create_status"
        title="Status"
      >
        Status
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="plans_create_status"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Pending"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
